### PR TITLE
EES-4642 Default and max limit for LT lists

### PR DESCRIPTION
**Pull request by vgarvardt**
Do not allow unlimited lists by default.

### CODE REVIEW

**Comment by diegomarangoni** [comment]
Would be nice to have this specified on the documentation

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -34,16 +34,18 @@ var (
...
 type Proxy struct {
 	maxLoadTestsRun int
+	maxListLimit    int64
 	registry        backends.Registry
 	kubeClient      *kube.Client
 }
 
 // NewProxy returns new Proxy handlers
-func NewProxy(maxLoadTestsRun int, registry backends.Registry, kubeClient *kube.Client) *Proxy {
+func NewProxy(maxLoadTestsRun int, registry backends.Registry, kubeClient *kube.Client, maxListLimit int64) *Proxy {
```
Comment (diegomarangoni) [inline pkg/proxy/proxy.go:43]:
```suggestion
func NewProxy(maxLoadTestsRun int, maxListLimit int64, registry backends.Registry, kubeClient *kube.Client) *Proxy {
```
what do you think about grouping parameters with same name?

File: pkg/proxy/request.go [hunk 2]
```diff
@@ -74,6 +76,10 @@ func fromHTTPRequestToListOptions(r *http.Request) (*kubernetes.ListOptions, err
 			return nil, err
 		}
 
+		if limit > maxListLimit {
+			return nil, fmt.Errorf("limit value is too big, max possible value is %d", maxListLimit)
```
Comment (s-radyuk) [inline pkg/proxy/request.go:80]: Let's add a test for this case to pkg/proxy/proxy_test.go

File: pkg/proxy/proxy.go (hunk 1 above)
Comment (vgarvardt) [reply pkg/proxy/proxy.go:43]: I would group them by importance, tbh, but I did not want to change a lot - adding to the end is the simplest case.

File: pkg/proxy/request.go (hunk 2 above)
Comment (vgarvardt) [reply pkg/proxy/request.go:80]: done

**Comment by diegomarangoni** [comment]
Can you add this new env var to helm chart pls?

**Comment by vgarvardt** [comment]
@diegomarangoni this change has nothing to do with helm chart. Currently we do not have docs with all the available env vars. Will create a task to create it.

**Comment by diegomarangoni** [comment]
+1

**Pull request merged**